File: java-solid-principle-sample-code/src/com/principle/solid/dependency_inversion/DIPMain.java
package com.principle.solid.dependency_inversion;

import java.util.Arrays;
import java.util.List;

public class DIPMain {

	public static void main(String[] args) {
		System.out.println("Hello World");
		
		Square square = new Square(10);
		
		List<Shape> shapes = Arrays.asList(square);
		
		IAreaCalculator areaCalculator = new AreaCalculator();		
		/**	Dependency Inversion Achieved - Instead of Concreted Class take a Interface Reference and Inject on-demand. */
		ShapePrinter shapePrinter = new ShapePrinter(areaCalculator);
		System.out.println(shapePrinter.json(shapes));
		System.out.println(shapePrinter.csv(shapes));
		
		areaCalculator = new AreaCalculatorV2();
		/**	Dependency Inversion Achieved - Instead of Concreted Class take a Interface Reference and Inject on-demand. */
		shapePrinter = new ShapePrinter(areaCalculator);
		System.out.println(shapePrinter.json(shapes));
		System.out.println(shapePrinter.csv(shapes));
	}
}

File: python-solid-principle-sample-code/com/principle/solid/dependency_inversion/DIPMain.py
from typing import List

from Shape import Shape
from Square import Square
from AreaCalculator import AreaCalculator
from AreaCalculatorV2 import AreaCalculatorV2
from ShapePrinter import ShapePrinter
from IAreaCalculator import IAreaCalculator


def main():
    print('Hello World')

    square: Square = Square(10)

    shapes: List[Shape] = [square]

    area_calculator: IAreaCalculator = AreaCalculator()
    """ Dependency Inversion Achieved - Instead of Concreted Class take a Interface Reference and Inject on-demand. """
    shape_printer: ShapePrinter = ShapePrinter(area_calculator)
    print(shape_printer.json(shapes))
    print(shape_printer.csv(shapes))

    area_calculator: IAreaCalculator = AreaCalculatorV2()
    """ Dependency Inversion Achieved - Instead of Concreted Class take a Interface Reference and Inject on-demand. """
    shape_printer: ShapePrinter = ShapePrinter(area_calculator)
    print(shape_printer.json(shapes))
    print(shape_printer.csv(shapes))


if __name__ == "__main__":
    main()
